tests/test-update-refspec-checkpoint.c

```diff
@@ -1393,10 +1393,10 @@ test_update_refspec_checkpoint_ignore_remote (EosUpdaterFixture *fixture,
 typedef struct
 {
   /* Setup */
-  const gchar *src_ref;  /* (nullable) for default */
-  const gchar *tgt_ref;  /* (nullable) for default */
-  const gchar *sys_vendor;  /* (nullable) for default */
-  const gchar *product_name;  /* (nullable) for default */
+  const gchar *src_ref;
+  const gchar *tgt_ref;
+  const gchar *sys_vendor;
+  const gchar *product_name;
```

Comment:
I notice they should not be `NULL`, otherwise the test procedure will show the error: `GLib-FATAL-CRITICAL: g_regex_match_full: assertion 'string != NULL' failed`.
The `src_ref`, `tgt_ref`, `sys_vendor` and `product_name` will be [assigned with default values' addresses](https://github.com/endlessm/eos-updater/blob/14b5cc57ac6945c0c368babca5bb7058bd4d5ad8/tests/test-update-refspec-checkpoint.c#L1698-L1702) if they point to `NULL`:
```C
      tests[i].src_ref = tests[i].src_ref ? tests[i].src_ref : "os/eos/amd64/latest1";
      tests[i].tgt_ref = tests[i].tgt_ref ? tests[i].tgt_ref : "os/eos/amd64/latest2";
      tests[i].sys_vendor = tests[i].sys_vendor ? tests[i].sys_vendor : default_vendor;
      tests[i].product_name = tests[i].product_name ? tests[i].product_name : default_product;
      do_update_refspec_checkpoint (fixture, user_data, &tests[i], host_is_aarch64);
```

Reply:
But they are `NULL` up until the defaults are overwritten into the `tests` array. So if these struct members are accessed elsewhere in the code, they could be `NULL`. I think the `(nullable)` annotation is supposed to indicate that the member can _never_ be `NULL`, in a type-safety sense.

(sorry for interjecting! I hope I haven’t misunderstood what’s going on here — just catching up with bugmail)

Reply:
I think @pwithnall is correct here, however I think I'm going to merge it as is since it's just test code with many other implicit semantics and fully handling `(nullable)` would require another round of `NULL` handling in the factored out `do_update_refspec_checkpoint` helper.